src/components/PersonalInfoForm.js

```diff
@@ -0,0 +1,182 @@
+import { useState } from "react";
+import "./PersonalInfoForm.css";
+
+function PersonalInfoForm() {
+  const PHONE_NUMBER_LENGTH = 10;
+  const MAX_PHONE_PREFIX_LENGTH = 6;
+  const DEFAULT_INPUT_SIZE = 20;
+
+  const placeholders = {
+    name: "Name",
+    phone: "Phone",
+    email: "Email",
+  };
+
+  const [values, setValues] = useState({
+    name: "",
+    phonePrefix: "+",
+    phone: "",
+    email: "",
+  });
+  const [errors, setErrors] = useState({
+    name: "",
+    phone: "",
+    email: "",
+  });
+
+  //TODO: Fetch user information if it already exists
+
+  const validateValueChange = (valueType, value) => {
+    let newMessage = "";
+    switch (valueType) {
+      case "phone":
+        if (value.length !== PHONE_NUMBER_LENGTH) {
+          newMessage = "Please enter a valid phone number!";
+        }
+        break;
+      case "phonePrefix":
+        const correctCountryCodeFormat = /^\+(\d{1}-)?(\d{1,3})$/;
+        if (!value.match(correctCountryCodeFormat)) {
+          newMessage = "Please enter a valid country code!";
+        }
+        break;
+      case "email":
+        if (
+          !new RegExp(/^([a-zA-Z0-9._%-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,})$/).test(
```

Comment:
Hmmm... are we sure that this will work? I'm not sure if we should use such hardcoded regex

Reply:
Is there another alternative that you think is better? Originally I was using the input of `type="email"` as a way to check but I got rid of the submit button in the form that checks that just because I wanted a more customizable approach to the form.

Reply:
Mainly I am concerned that as a reviewer I not sure if this regex works correctly. Maybe, you can add unit tests to show that this works as expected. Or, you can use some existing library that provides this regex to be used